hooks/use-dappnode-deposit.ts

```diff
@@ -0,0 +1,237 @@
+import { useCallback, useState } from "react";
+import {
+  useAccount,
+  useReadContract,
+  useWriteContract,
+  useWaitForTransactionReceipt,
+} from "wagmi";
+import CONTRACTS, { ContractNetwork } from "@/utils/contracts";
+import dappnodeIncentiveABI from "@/utils/abis/dappnodeIncentive";
+import { loadCachedDeposits } from "@/utils/deposit";
+import { getPublicClient } from "wagmi/actions";
+import { config } from "@/wagmi";
+import { fetchDeposit } from "@/utils/fetchEvents";
+
+export type DepositDataJson = {
+  pubkey: string;
+  withdrawal_credentials: string;
+  amount: bigint;
+  signature: string;
+  deposit_message_root: string;
+  deposit_data_root: string;
+  fork_version: string;
+};
+
+export type DappnodeUser = [
+  safe: string,
+  status: number,
+  expectedDepositCount: number, // uint16
+  totalStakeAmount: bigint // uint256
+];
+
+function useDappnodeDeposit(contractConfig: ContractNetwork | undefined, address: `0x${string}` | undefined, chainId: number) {
+  const [deposits, setDeposits] = useState<DepositDataJson[]>([]);
+  const [hasDuplicates, setHasDuplicates] = useState(false);
+  const [isBatch, setIsBatch] = useState(false);
+  const [filename, setFilename] = useState("");
+  const client = getPublicClient(config, { chainId: chainId as 100 });
+
+  const { data: user }: { data: DappnodeUser | undefined } = useReadContract({
+    abi: dappnodeIncentiveABI,
+    address: contractConfig?.addresses.dappnodeIncentive,
+    functionName: "users",
+    args: [address],
+  });
+
+  const isWrongNetwork = chainId !== 100;
+  const { data: depositHash, writeContractAsync, isPending, isError } = useWriteContract();
+  const { isSuccess: depositSuccess } = useWaitForTransactionReceipt({
+    hash: depositHash,
+  });
+
+  const dappnodeValidate = useCallback(
+    async (deposits: DepositDataJson[]) => {
+      let newDeposits = [];
+      let hasDuplicates = false;
+      let _isBatch = false;
+      if (contractConfig && user) {
+        const checkJsonStructure = (depositDataJson: DepositDataJson) => {
+          return (
+            depositDataJson.pubkey &&
+            depositDataJson.withdrawal_credentials &&
+            depositDataJson.amount &&
+            depositDataJson.signature &&
+            depositDataJson.deposit_message_root &&
+            depositDataJson.deposit_data_root &&
+            depositDataJson.fork_version
+          );
+        };
```

Comment:
A bit repetitive,  maybe some functionality can be shared with `use-deposit'?

Reply:
I agree too. Also on the UI, i think it makes more sense to move the dappnode deposit component within the basic deposit and have a switcher to on/off dappnode mode.